test/e2e/scoped_client_test.go

```diff
@@ -87,67 +83,55 @@ var _ = Describe("Scoped Client", func() {
 		// 5. Invoke Get API call on non existent object(s) to check if
 		//    the call can be made successfully.
 		namespace := genName("a")
-		_, cleanupNS := newNamespace(GinkgoT(), kubeclient, namespace)
+		_, cleanupNS := newNamespace(kubeclient, namespace)
 		defer cleanupNS()
 
 		saName := genName("user-defined-")
-		sa, cleanupSA := newServiceAccount(GinkgoT(), kubeclient, namespace, saName)
+		sa, cleanupSA := newServiceAccount(kubeclient, namespace, saName)
 		defer cleanupSA()
 
-		waitForServiceAccountSecretAvailable(GinkgoT(), kubeclient, sa.GetNamespace(), sa.GetName())
+		By("Wait for ServiceAccount secret to be available")
+		Eventually(func() (bool, error) {
+			sa, err := kubeclient.KubernetesInterface().CoreV1().ServiceAccounts(sa.GetNamespace()).Get(context.TODO(), sa.GetName(), metav1.GetOptions{})
+			Expect(err).ShouldNot(HaveOccurred())
+			if len(sa.Secrets) > 0 {
```

Comment:
Could put these 2 in the same if statement. there is also no error thrown (other than timeout) if there's no secret in the SA.

Reply:
Clarified it offline. The current implementation seems reasonable